### PR TITLE
Mm 40933 run retrospective edit enter run metrics backend

### PR DESCRIPTION
**Pull request by gbochora**
#### Summary
This PR will:
- Add storing and retrieving playbook run metrics values.
- Add a new field to the `PlaybookRun` struct for storing/retrieving metrics data.
- Modify UpdateRetrospective/PublishRetrospective functions signatures, now they take RetrospectiveUpdate struct as an argument.

Notes:
Each time a user navigates to the run backstage page, we retrieve the playbook and run objects from the backend. Based on that:
- I've not added metric configuration info to the `PlaybookRun` struct. Instead, we can get it from the playbook object.
- I've added metrics data only to the `GetPlaybookRun` method. `GetPlaybookRuns` method will return run objects without metrics data.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40933

#### Checklist

- ~~[ ] Telemetry updated~~
- ~~[ ] Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

**Review by agarciamontoro — APPROVED** [review]
Looks good! I was a bit confused at first between the metrics configuration and the metrics data, but I think I got it after looking at your previous PR!
I left a small nit in the tests, but nothing blocking at all. Great work :tada:

File: server/sqlstore/playbook_run_test.go [hunk 1]
```diff
@@ -260,6 +309,50 @@ func TestUpdatePlaybookRun(t *testing.T) {
...
+
+		// delete one metric config from playbook
+		playbook.Metrics = playbook.Metrics[1:]
+		err = playbookStore.Update(playbook)
+		require.NoError(t, err)
+
+		// should return single metric
+		actual, err := playbookRunStore.GetPlaybookRun(playbookRun.ID)
+		require.NoError(t, err)
+		require.Equal(t, len(actual.MetricsData), 1)
```
[the remark this answers is not in the document]
Comment (gbochora) [reply server/sqlstore/playbook_run_test.go:351]: Thanks : )

File: server/app/playbook_run_service.go [hunk 2]
```diff
@@ -2384,12 +2386,12 @@ func (s *PlaybookRunServiceImpl) PublishRetrospective(playbookRunID, text, publi
...
 
 	retrospectiveURL := getRunRetrospectiveURL("", playbookRunToPublish.ID)
-	if _, err = s.poster.PostMessage(playbookRunToPublish.ChannelID, "@channel Retrospective has been published by @%s\n[See the full retrospective](%s)\n%s", publisherUser.Username, retrospectiveURL, text); err != nil {
+	if _, err = s.poster.PostMessage(playbookRunToPublish.ChannelID, "@channel Retrospective has been published by @%s\n[See the full retrospective](%s)\n%s", publisherUser.Username, retrospectiveURL, retrospective.Text); err != nil {
 		return errors.Wrap(err, "failed to post to channel")
 	}
 
 	telemetryString := fmt.Sprintf("?telem_action=follower_clicked_retrospective_dm&telem_run_id=%s", playbookRunToPublish.ID)
-	retrospectivePublishedMessage := fmt.Sprintf("@%s published the retrospective report for [%s](%s%s).\n%s", publisherUser.Username, playbookRunToPublish.Name, retrospectiveURL, telemetryString, text)
+	retrospectivePublishedMessage := fmt.Sprintf("@%s published the retrospective report for [%s](%s%s).\n%s", publisherUser.Username, playbookRunToPublish.Name, retrospectiveURL, telemetryString, retrospective.Text)
```
Comment (cpoile) [inline server/app/playbook_run_service.go:2394]: I'm assuming the custom post will be updated later with the metrics values, we're just keeping it the same for now, correct?

**Review by cpoile — APPROVED** [review]
Nice work Giorgi!

File: server/app/playbook_run_service.go (hunk 2 above)
Comment (gbochora) [reply server/app/playbook_run_service.go:2394]: Yes, that's correct.

**Pull request merged**